teraserver/python/modules/FlaskModule/API/service/ServiceQueryParticipantGroups.py

```diff
@@ -0,0 +1,179 @@
+        if not participant_group_info['participant_group_name']:
+            return gettext('Invalid participant group name'), 403
+
+        # Initialize participant group instance
+        participant_group: TeraParticipantGroup = TeraParticipantGroup()
+
+        # Check if it's a new participant group or an update
+        if participant_group_info['id_participant_group'] == 0:
+            # Create participant group
+            participant_group = TeraParticipantGroup()
+            participant_group.participant_group_name = participant_group_info['participant_group_name']
+            participant_group.id_project = participant_group_info['id_project']
+
+            try:
+                TeraParticipantGroup.insert(participant_group)
+            except IntegrityError as e:
+                self.module.logger.log_warning(self.module.module_name, ServiceQueryParticipantGroups.__name__,
+                                               'insert',
+                                               400, 'Integrity error', str(e))
+
+                # Handle integrity error related to projects
+                if 't_projects' in str(e.args):
+                    return gettext('Can\'t insert participant group: participant group\'s project '
+                                   'is disabled or invalid.'), 400
+        else:
+            # Update existing participant group
+            try:
+                TeraParticipantGroup.update(participant_group_info['id_participant_group'], participant_group_info)
+            except IntegrityError as e:
+                self.module.logger.log_warning(self.module.module_name, ServiceQueryParticipantGroups.__name__,
+                                               'update',
+                                               400, 'Integrity error', str(e))
+
+                # Handle integrity error related to projects
+                if 't_projects' in str(e.args):
+                    return gettext('Can\'t update participant group: participant group\'s project is disabled.'), 400
+
+            # Retrieve the updated participant group
+            participant_group = TeraParticipantGroup.get_participant_group_by_id(participant_group_info
+                                                                                 ['id_participant_group'])
+
+        # Return the JSON representation of the participant group
+        return participant_group.to_json(minimal=False)
+
+
+
+    @api.doc(description='Delete a specific participant group.',
+             responses={200: 'Success',
+                        403: 'Logged user doesn\'t have permission to access the requested data',
+                        500: 'Database error.'},
+             params={'token': 'Secret token'})
+    @api.expect(delete_parser)
+    @LoginModule.service_token_or_certificate_required
+    def delete(self):
+        args = delete_parser.parse_args()
+        id_todel = args['id']
+

```

Comment:
Add security check (see above) with service_access (validating the `id` is in the returned list of `get_accessible_participant_groups` method.